### PR TITLE
ENH: update watershed API

### PR DESCRIPTION
**Pull request by ambrosejcarr**
* improve typing
* f-style strings
* `segment` becomes `run`

### CODE REVIEW

File: starfish/pipeline/segmentation/__init__.py [hunk 1]
```diff
@@ -45,18 +47,18 @@ def _cli(cls, args, print_help=False):
...
         with open(args.output, "w") as f:
             f.write(json.dumps(geojson))
 
 
-def regions_to_geojson(r, use_hull=True):
+def regions_to_geojson(r, use_hull=True) -> List[Dict[str, Dict[str, Any]]]:
     """Convert region geometrical data to geojson format"""
 
-    def make_dict(id_, verts):
+    def make_dict(id_, verts) -> Dict[str, Dict[str, Any]]:
```
Comment (ttung) [inline starfish/pipeline/segmentation/__init__.py]: what's the types of id_ and verts?

File: starfish/stats.py [hunk 2]
```diff
@@ -1,36 +1,40 @@
...
 from regional import many as Many
 from regional import one as One
 from scipy.sparse import coo_matrix
 
+from starfish.image import ImageStack
+from starfish.types import Number
 
-def stack_describe(stack):
+
+def stack_describe(stack) -> List[Dict[str, Number]]:
```
Comment (ttung) [inline starfish/stats.py]: stack: ImageStack

File: starfish/stats.py [hunk 3]
```diff
@@ -1,36 +1,40 @@
...
     num_rounds = stack.shape[0]
     stats = [im_describe(stack[k, :]) for k in range(num_rounds)]
     return stats
 
 
-def im_describe(im):
+def im_describe(im) -> Dict[str, Number]:
```
Comment (ttung) [inline starfish/stats.py]: im: np.ndarray

File: starfish/stats.py [hunk 4]
```diff
@@ -1,36 +1,40 @@
...
     shape = im.shape
     flat_dims = reduce(lambda x, y: x * y, shape)
     flat_im = np.reshape(im, flat_dims)
     stats = pd.Series(flat_im).describe()
     return stats.to_dict()
 
 
-def label_to_regions(labels):
+def label_to_regions(labels) -> Many:
```
Comment (ttung) [inline starfish/stats.py]: what's labels?

File: starfish/stats.py [hunk 5]
```diff
@@ -40,18 +44,23 @@ def region_for(label_mat_coo, label):
     return Many(regions)
 
 
-def measure(im, labels, num_objs, measurement_type='mean'):
+def measure(
+        im: np.ndarray, labels: Sequence[Number], num_objs: int, measurement_type: str='mean') \
```
Comment (ttung) [inline starfish/stats.py]: no backslash.

File: starfish/stats.py [hunk 6]
```diff
@@ -40,18 +44,23 @@ def region_for(label_mat_coo, label):
...
-        raise ValueError('Unsporrted measurement type: {}'.format(measurement_type))
+        raise ValueError(f'Unsupported measurement type: {measurement_type}. Choose one of "mean" '
+                         f'or "max".')
 
     return res
 
 
-def measure_stack(stack, labels, num_objs, measurement_type='mean'):
+def measure_stack(
+        stack: ImageStack, labels: Sequence[Number], num_objs: int, measurement_type='mean') -> \
```
Comment (ttung) [inline starfish/stats.py]: no backslash.

File: starfish/pipeline/segmentation/watershed.py [hunk 7]
```diff
@@ -1,11 +1,13 @@
 import numpy as np
 import scipy.ndimage.measurements as spm
+from regional import many
```
Comment (ttung) [inline starfish/pipeline/segmentation/watershed.py]: nit: prefer import regional, and then using `regional.many`, like in https://github.com/spacetx/starfish/pull/368/files#diff-54c1560b3315fe0c319c3bebcad83299

rationale: if it's not obvious what package something comes from (i.e., not a pretty standard library), I prefer a more qualified name.

File: starfish/pipeline/segmentation/__init__.py (hunk 1 above)
Comment (ambrosejcarr) [reply starfish/pipeline/segmentation/__init__.py]: I think this is a @dganguli question.

File: starfish/stats.py (hunk 4 above)
Comment (ambrosejcarr) [reply starfish/stats.py]: Also an @dganguli question.

**Pull request merged**